app/models/package_manager/maven.rb

```diff
@@ -153,7 +163,10 @@ def self.retrieve_versions(versions, name)
     end
 
     def self.download_pom(group_id, artifact_id, version)
-      pom_request = request(MavenUrl.new(group_id, artifact_id, repository_base).pom(version))
+      url = MavenUrl.new(group_id, artifact_id, repository_base).pom(version)
+      pom_request = request(url)
+      raise POMNotFound.new(url) if pom_request.status == 404
```

Comment:
Will we have any useful information if we can't get the pom?

Reply:
Unfortunately not. We can only rely on maven_metadata.xml for getting the list of versions, and we get the rest of the data in `mapping(raw_project)` from the latest_version's POM. 

I guess we could iterate thru each version and find the newest one with a POM, but that wouldn't work in this specific Spring example because none of the versions of a POM.